pkg/router/routeapihelpers/validation.go

```diff
@@ -353,6 +363,70 @@ func validateInsecureEdgeTerminationPolicy(tls *routev1.TLSConfig, fldPath *fiel
 	return nil
 }
 
+// isSelfSignedCert determines if a certificate is self-signed by verifying that the issuer matches the subject,
+// the authority key identifier matches the subject key identifier, and the public key algorithm matches the
+// signature algorithm. This logic mirrors the approach that OpenSSL uses to set the EXFLAG_SS flag, which
+// indicates a certificate is self-signed.
+// Ref: https://github.com/openssl/openssl/blob/b85e6f534906f0bf9114386d227e481d2336a0ff/crypto/x509/v3_purp.c#L557
+func isSelfSignedCert(cert *x509.Certificate) bool {
+	isIssuerEqualToSubject := bytes.Equal(cert.RawIssuer, cert.RawSubject)
+	isAuthorityKeyEqualToSubjectKey := bytes.Equal(cert.AuthorityKeyId, cert.SubjectKeyId)
+	isAlgorithmConsistent := signatureAlgorithmToPublicKeyAlgorithm(cert.SignatureAlgorithm) == cert.PublicKeyAlgorithm
+	return isIssuerEqualToSubject &&
+		(cert.AuthorityKeyId == nil || isAuthorityKeyEqualToSubjectKey) &&
+		isAlgorithmConsistent
```

Comment:
This is a nit.  Change if there ends up being other things to change. The variable names could be a little more clear since they are bool:
```suggestion
	issuerIsEqualToSubject := bytes.Equal(cert.RawIssuer, cert.RawSubject)
	authorityKeyIsEqualToSubjectKey := bytes.Equal(cert.AuthorityKeyId, cert.SubjectKeyId)
	algorithmIsConsistent := signatureAlgorithmToPublicKeyAlgorithm(cert.SignatureAlgorithm) == cert.PublicKeyAlgorithm
	return issuerIsEqualToSubject &&
		(cert.AuthorityKeyId == nil || authorityKeyIsEqualToSubjectKey) &&
		algorithmIsConsistent
```